java/src/main/native/src/row_conversion.hpp

```diff
@@ -25,12 +25,23 @@
 namespace cudf {
 namespace java {
 
+std::vector<std::unique_ptr<cudf::column>> convert_to_rows_fixed_width_optimized(
```

Comment:
Docs for this and `convert_from_rows_fixed_width_optimized`.   What are the use cases here where we switch between the regular and "optimized" versions?